_posts/2025-05-07-release-notes-3.7.0.md

```diff
@@ -0,0 +1,458 @@
+    createdAt: Timestamp,
+    shippedAt: Option[Timestamp] = None,
+    deliveredAt: Option[Timestamp] = None,
+    itemIds: List[ItemId] = Nil
+  )
+  
+  val orders = List.empty[Order]
+  val selected = orders.collect:
+    case Order(id = s"special-${internalId}", shippedAt = None) => internalId
+    case Order(id = orderId, itemIds = items) if items.size > 5 => orderId
+```
+
+Last, but not least, named tuples are opening a new paradigm of metaprogramming by letting you compute structural types without the need for macros!
+The `Selectable` trait now has a `Fields` type member that can be instantiated to a named tuple.
+
+```scala
+class QueryResult[T](rawValues: Map[String, Any]) extends Selectable:
+  type Fields = NamedTuple.Map[NamedTuple.From[T], Option]
+  def selectDynamic(fieldName: String) = rawValues.get(fieldName)
+  
+case class Release(version: String, issues: List[String])
+
+@main def Test =
+  val query: QueryResult[Release] = QueryResult(Map("version" -> "3.7.0"))
+  val version: Option[String] = query.version
+  assert(query.version.contains("3.7.0"))
+  assert(query.issues.isEmpty)
+```
+
+## Preview SIPs
+
+### Preview features
+
+Scala 3.7 introduces the concept of `preview` features — fully implemented and SIP-approved, but potentially subject to small refinements.
+New Scala language features or standard library APIs are initially introduced as experimental, but once they become fully implemented and accepted by the [SIP](https://docs.scala-lang.org/sips/) these can become a `preview` feature.
+
+Preview language features and APIs are guaranteed to be standardized in some next Scala minor release, but allow the compiler team to introduce small, possibly binary incompatible, changes based on the community feedback.
+These can be used by early adopters who can accept the possibility of binary compatibility breakage. For instance, preview features could be used in some internal tool or application. On the other hand, preview features are discouraged in publicly available libraries.
+
+All preview features can be enabled using `-preview` flag passed to the compiler. More information about preview features can be found in [preview definitions guide](https://docs.scala-lang.org/scala3/reference/preview/overview.html)
+
+### [SIP-62: For comprehension improvements](https://docs.scala-lang.org/sips/better-fors.html)
+
+The first introduced preview feature is [SIP-62 - `Better Fors`](https://docs.scala-lang.org/sips/better-fors.html) originally introduced in Scala 3.6 as an experimental.
+A major user-facing improvement introduced by this SIP is the ability to start a for-comprehension block with aliases.
+It also introduces changes to how your code is desugared by the compiler, leading to a more optimized code by removing some redundant calls.
+
+```scala
+//> using options -preview
+//> using scala 3.7
+@main def betterFors = 
+  for
+    a = 1
+    b <- Some(2)
+    c <- Option.when(a < 5)(10)
+  yield b * c
+```
```

Comment:
It's clear that that this example shows that you can now start with `a = 1`, but it isn't clear what _else_ it is intended to demonstrate. I'd suggest adding a sentence such as "This code used to result in a redundant extra call to..."

Reply:
I've added the extra example showing the previous wrapping results into tuples as intermiediete step